packages/electron-builder-squirrel-windows/src/SquirrelWindowsTarget.ts

```diff
@@ -42,12 +66,21 @@ export default class SquirrelWindowsTarget extends Target {
     this.appDirectory = appOutDir
     this.outputDirectory = installerOutDir
     const distOptions = await this.computeEffectiveDistOptions()
-    if (distOptions.vendorDirectory) {
-      this.select7zipArch(distOptions.vendorDirectory, arch)
-    }
 
+    distOptions.vendorDirectory = await this.prepareSignedVendorDirectory()
+    this.select7zipArch(distOptions.vendorDirectory, arch)
+    distOptions.fixUpPaths = true
+    distOptions.setupExe = setupFile
+    if (this.options.msi) {
+      distOptions.setupMsi = setupFile.replace(".exe", ".msi")
+    }
     await createWindowsInstaller(distOptions)
 
+    await packager.signAndEditResources(artifactPath, arch, installerOutDir)
+    if (this.options.msi && process.platform === "win32") {
+      await packager.sign(artifactPath.replace(".exe", ".msi"))
+    }
```

Comment:
Is this intentional? We only sign when `this.options.msi && process.platform === "win32"` but we call `callArtifactBuildCompleted` only if `this.options.msi`
Why the additional `process.platform === 'win32'` check?

Nit: We also call `artifactPath.replace(".exe", ".msi")` twice, could be worth pulling out into a `const msiArtifactPath =`

Reply:
> Why the additional process.platform === 'win32' check?

It's a typo. Fixed.

> Nit: We also call artifactPath.replace(".exe", ".msi") twice, could be worth pulling out into a const msiArtifactPath =

Fixed.